test/test_nestedtensor.py

```diff
@@ -549,6 +552,109 @@ def test_nested_tensor_indexing(self, device, dtype):
+        output = x[1:]
+        for i in range(1, x.size(0)):
+            self.assertEqual(x[i], output[i - 1])
+
+        output = x[:2]
+        for i in range(0, 2):
+            self.assertEqual(x[i], output[i])
+
+        # Success case: slicing regular dimension
+        NUM_CHUNKS = 4
+        CHUNK_SIZE = x.size(-1) // NUM_CHUNKS
+        output = x.chunk(NUM_CHUNKS, dim=2)
+        self.assertEqual(len(output), NUM_CHUNKS)
+        for chunk_idx, nt_chunk in enumerate(output):
+            self.assertTrue(nt_chunk.is_nested)
+
+            # Ensure same number of underlying tensor components
+            self.assertEqual(nt_chunk.dim(), x.dim())
+            self.assertEqual(nt_chunk.dim(), x.dim())
+
+            for i, component in enumerate(nt_chunk):
+                # Compare to manual slicing
+                start = chunk_idx * CHUNK_SIZE
+                end = start + CHUNK_SIZE
+                expected = x[i][:, start:end]
+                self.assertEqual(component, expected)
+
+        # Success case: chunking empty NT results in N chunks of original shape
+        empty = torch.nested_tensor([
+            torch.randn(0, 2, device=device, dtype=dtype),
+            torch.randn(0, 3, device=device, dtype=dtype),
+        ])
+        output = empty.chunk(NUM_CHUNKS, dim=1)
+        self.assertEqual(len(output), NUM_CHUNKS)
+        for nt_chunk in output:
+            self.assertEqual(nt_chunk.numel(), 0)
+            for i, component in enumerate(nt_chunk):
+                self.assertEqual(empty[i].shape, component.shape)
+
+        # Error case: slicing jagged dimension
+        with self.assertRaisesRegex(RuntimeError, 'dimension 1 is irregular'):
+            x.narrow(1, 1, 2)
+
+        with self.assertRaisesRegex(RuntimeError, 'dimension 1 is irregular'):
+            x.chunk(2, dim=1)
+
+        # Errors when calling slice.backwards()
+        a = torch.randn(3, 3 * 4, device=device, dtype=dtype, requires_grad=True)
+        b = torch.randn(2, 3 * 4, device=device, dtype=dtype, requires_grad=True)
+        nt_grad = torch.nested_tensor([a, b])
+        output = nt_grad[1:]
+        self.assertRaisesRegex(NotImplementedError, "the derivative for 'slice' is not implemented.",
+                               lambda: output.backward(output.clone()))
+
+    @dtypes(*floating_types_and_half())
+    def test_nested_tensor_chunk(self, device, dtype):
+        # transformer case
+        a = torch.randn(3, 3 * 4, device=device, dtype=dtype)
+        b = torch.randn(2, 3 * 4, device=device, dtype=dtype)
+        c = torch.randn(1, 3 * 4, device=device, dtype=dtype)
+        a_chunks = a.chunk(3, dim=-1)
+        b_chunks = b.chunk(3, dim=-1)
+        c_chunks = c.chunk(3, dim=-1)
+
+        a_nt = [a_chunks[0], b_chunks[0], c_chunks[0]]
+        b_nt = [a_chunks[1], b_chunks[1], c_chunks[1]]
+        c_nt = [a_chunks[2], b_chunks[2], c_chunks[2]]
+
+        nt = torch.nested_tensor([a, b, c])
```

Comment:
@drisspg This might landrance with the nested_tensor/as_nested_tensor PR )":

Would you be open to waiting for that to land before landing this? Asking because I'm landing that via phabricator due to internal changes that will break tests and phabricator tests take extremely long to run (and are almost done for the latest rebase right now!)

Alternatively, could we land this test case specifically in a separate PR (I could open a PR that adds this test back after my PR lands)? If not I'll wait for this to land and then rebase on my end (but just a little nervous about getting everything in hahah)

Reply:
Yeah of course, I just canceled my merge job. Just Ping me and let mw know when I can land. I assume it will be tomorrow after the diff train with yours lands?

Reply:
hm mine is not in diff train bcos i'm clicking land on phabricator! If nothing goes wrong it should be landed before tmr morning, will ping you!